src/main/java/org/terasology/wildAnimals/system/WildAnimalsSpawnSystem.java

```diff
@@ -116,7 +107,14 @@ public void onChunkGenerated(OnChunkGenerated event, EntityRef worldEntity) {
             return;
         }
         Vector3i chunkPos = JomlUtil.from(event.getChunkPos());
-        tryDeerSpawn(chunkPos);
+        // randomly decide whether to spawn deer or sheep in this chunk
+        Prefab animalPrefab;
+        if (random.nextInt(2) % 2 == 0) {
+            animalPrefab = Assets.getPrefab("WildAnimals:Deer").get();
+        } else {
+            animalPrefab = Assets.getPrefab("WildAnimals:Sheep").get();
+        }
```

Comment:
Instead of doing the asset lookup each time a chunk is being generated this my be done in `initialise()` as before. Instead of keeping track of individual prefabs you could fill a single `List<Prefab>` of flock animals:

```java
private List<Prefab> flockAnimals;
// ...
flockAnimals = new ArrayList<>();
Assets.getPrefab("WildAnimals:Deer").ifPresent(prefab -> flockAnimals.add(prefab));
Assets.getPrefab("WildAnimals:Sheep").ifPresent(prefab -> flockAnimals.add(prefab));
```

Selecting a random animal from that list could then be done like this (not sure whether it's `.size()` or `.length()`):
```suggestion
        Prefab animalPrefab = flockAnimals.get(random.nextInt(flockAnimals.size()));
```